### PR TITLE
AG-17235 Auto-generate column definitions from row data

### PR DESCRIPTION
**Pull request by StephenCooper**
## Summary

- Adds a new `autoGenerateColumnDefs` grid option that automatically generates column definitions by inspecting `rowData` keys, removing the need to manually define `columnDefs` for simple use cases
- Introduces `processAutoGeneratedColumnDefs` callback for customising the generated columns (renaming headers, setting widths, filtering which columns appear, etc.)
- Implements as a new `AutoGenerateColumnsModule` community module 
- Adds documentation page with three examples: basic usage, customising columns, and nested objects
- Includes comprehensive behavioural test suite (955 lines) covering edge cases like dynamic row data updates, nested objects, null/undefined handling, and interaction with existing `columnDefs`

Fix [AG-17235](https://ag-grid.atlassian.net/browse/AG-17235)

A File Drop overlay will follow in another PR

### CODE REVIEW

File: packages/ag-grid-community/src/autoGenerateColumns/autoGenerateColumnsService.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import type { AutoGenerateColumnDefsConfig } from '../entities/gridOptions';
+
+export class AutoGenerateColumnsService extends BeanStub implements NamedBean {
+    beanName = 'autoGenColsSvc' as const;
+
+    public postConstruct(): void {
+        const generateColsFunc = () => this.generateColumns();
+        generateColsFunc();
+        this.addManagedPropertyListener('rowData', generateColsFunc);
+        this.addManagedPropertyListener('autoGenerateColumnDefs', generateColsFunc);
```
[the remark this answers is not in the document]
Comment (StephenCooper) [reply packages/ag-grid-community/src/autoGenerateColumns/autoGenerateColumnsService.ts]: No, this is not required. The callback can just be used on the next run.

File: packages/ag-grid-community/src/autoGenerateColumns/autoGenerateColumnsService.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        const optionValue = gos.get('autoGenerateColumnDefs');
+        if (!optionValue) {
+            return;
+        }
+
+        const rowData = gos.get('rowData');
+        if (!rowData || rowData.length === 0) {
+            return;
+        }
+
```
[the remark this answers is not in the document]
Comment (StephenCooper) [reply packages/ag-grid-community/src/autoGenerateColumns/autoGenerateColumnsService.ts:25]: Fixed.

File: documentation/ag-grid-docs/src/content/docs/auto-generate-columns/_examples/configuration/index.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<div class="example-wrapper">
+    <div class="example-header">
+        <label>
+            objectValues:
+            <select id="objectValues" onchange="onObjectValues()">
```
[the remark this answers is not in the document]
Comment (StephenCooper) [reply documentation/ag-grid-docs/src/content/docs/auto-generate-columns/_examples/configuration/index.html:5]: fixed

File: packages/ag-grid-community/src/autoGenerateColumns/autoGenerateColumnsService.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        const rowData = gos.get('rowData');
+        if (!rowData || rowData.length === 0) {
+            if (!isInitial) {
+                gos.updateGridOptions({ options: { columnDefs: [] }, source: 'optionsUpdated' });
+            }
+            return;
+        }
+
+        const config: AutoGenerateColumnDefsOptions = typeof optionValue === 'object' ? optionValue : {};
```
[the remark this answers is not in the document]
Comment (StephenCooper) [reply packages/ag-grid-community/src/autoGenerateColumns/autoGenerateColumnsService.ts:26]: Not going to provide support for classes inside of RowData

**Review by SalvatorePreviti — APPROVED** [review]
